### PR TITLE
Specify Notary in `CashIssueAndPaymentFlow` and `PaymentRequest`

### PR DESCRIPTION
**Pull request by thschroeter**
Ensure the explorer works with networks with multiple notaries.

### CODE REVIEW

File: finance/src/main/kotlin/net/corda/finance/flows/CashPaymentFlow.kt [hunk 1]
```diff
@@ -32,14 +32,17 @@ open class CashPaymentFlow(
...
+        val issuerConstraint: Set<Party> = emptySet(),
+        val notary: Party? = null) : AbstractCashFlow<AbstractCashFlow.Result>(progressTracker) {
     /** A straightforward constructor that constructs spends using cash states of any issuer. */
     constructor(amount: Amount<Currency>, recipient: Party) : this(amount, recipient, true, tracker())
 
     /** A straightforward constructor that constructs spends using cash states of any issuer. */
     constructor(amount: Amount<Currency>, recipient: Party, anonymous: Boolean) : this(amount, recipient, anonymous, tracker())
 
-    constructor(request: PaymentRequest) : this(request.amount, request.recipient, request.anonymous, tracker(), request.issuerConstraint)
+    constructor(amount: Amount<Currency>, recipient: Party, anonymous: Boolean, notary: Party) : this(amount, recipient, anonymous, tracker(), notary=notary)
```
Comment (vkolomeyko) [inline finance/src/main/kotlin/net/corda/finance/flows/CashPaymentFlow.kt]: Minor: spacing around `notary=notary`

File: tools/explorer/src/main/kotlin/net/corda/explorer/views/cordapps/cash/NewTransaction.kt [hunk 2]
```diff
@@ -153,7 +153,7 @@ class NewTransaction : Fragment() {
             when (it) {
                 executeButton -> when (transactionTypeCB.value) {
                     CashTransaction.Issue -> IssueAndPaymentRequest(Amount.fromDecimal(amount.value, currencyChoiceBox.value), issueRef, partyBChoiceBox.value.party, notaries.first().value!!, anonymous)
-                    CashTransaction.Pay -> PaymentRequest(Amount.fromDecimal(amount.value, currencyChoiceBox.value), partyBChoiceBox.value.party, anonymous = anonymous)
+                    CashTransaction.Pay -> PaymentRequest(Amount.fromDecimal(amount.value, currencyChoiceBox.value), partyBChoiceBox.value.party, anonymous = anonymous, notary=notaries.first().value!!)
```
Comment (vkolomeyko) [inline tools/explorer/src/main/kotlin/net/corda/explorer/views/cordapps/cash/NewTransaction.kt]: Minor: I would create a local `val` for `notaries.first().value!!` to clearly show how we are selecting a Notary.

Comment (vkolomeyko) [reply tools/explorer/src/main/kotlin/net/corda/explorer/views/cordapps/cash/NewTransaction.kt]: Also, spacing around `notary=...`

**Comment by vkolomeyko** [comment]
In my view, this is reasonably small/isolated change which we may well incorporate into GA.
One worry I have is whether `CashPaymentFlow` remains binary compatible with those changes?

File: finance/src/main/kotlin/net/corda/finance/flows/CashPaymentFlow.kt [hunk 3]
```diff
@@ -32,14 +32,17 @@ open class CashPaymentFlow(
         val recipient: Party,
         val anonymous: Boolean,
         progressTracker: ProgressTracker,
-        val issuerConstraint: Set<Party> = emptySet()) : AbstractCashFlow<AbstractCashFlow.Result>(progressTracker) {
+        val issuerConstraint: Set<Party> = emptySet(),
+        val notary: Party? = null) : AbstractCashFlow<AbstractCashFlow.Result>(progressTracker) {
```
Comment (adagys) [inline finance/src/main/kotlin/net/corda/finance/flows/CashPaymentFlow.kt]: maybe better to fall back to the first notary inside the flow?

**Pull request merged**